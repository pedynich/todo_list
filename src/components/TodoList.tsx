import TodoItem from './TodoItem';
import { useAppSelector } from '../hook';
import AllToggle from './AllToggle';

const TodoList: React.FC = () => {
  const todos = useAppSelector(state => state.todos.list);

  return (
    <section className='main'>
      <AllToggle />
      <ul className='todo-list'>
        {todos.map((todo) => (
          <TodoItem
            key={todo.id}
            {...todo}
          />
        ))}
      </ul>
    </section>
  );
};

export default TodoList;