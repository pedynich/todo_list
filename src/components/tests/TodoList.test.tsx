import TodoList from "../TodoList";
import * as reduxHooks from '../../hook';
import { render } from "@testing-library/react";

jest.mock('react-redux')

const todos = [
    { id: '123', title: 'React', completed: false},
    { id: '124', title: 'Redux', completed: false}
];

describe('TodoList', () => {
    it('should create TodoList with empty todos', () => {
        jest.spyOn(reduxHooks, 'useAppSelector').mockReturnValue([])
        const view = render(<TodoList/>);

        expect(view).toMatchSnapshot();
    });

    it('should create TodoList with  todos items', () => {
        const dispatch = jest.fn()
        jest.spyOn(reduxHooks, 'useAppSelector').mockReturnValue(todos)
        jest.spyOn(reduxHooks, 'useAppDispatch').mockReturnValue(dispatch)
        const view = render(<TodoList/>);

        expect(view).toMatchSnapshot();
    })
})