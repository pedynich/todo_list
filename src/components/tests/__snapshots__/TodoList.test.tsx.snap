// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList should create TodoList with  todos items 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="main"
      >
        <div>
          <input
            class="toggle-all"
            id="toggle-all"
            type="checkbox"
          />
          <label
            for="toggle-all"
          >
            Mark all as complete
          </label>
        </div>
        <ul
          class="todo-list"
        >
          <li
            class=" "
          >
            <div
              class="view"
            >
              <input
                class="toggle"
                type="checkbox"
              />
              <label>
                React
              </label>
              <button
                class="destroy"
              />
            </div>
            <input
              class="edit"
              value="React"
            />
          </li>
          <li
            class=" "
          >
            <div
              class="view"
            >
              <input
                class="toggle"
                type="checkbox"
              />
              <label>
                Redux
              </label>
              <button
                class="destroy"
              />
            </div>
            <input
              class="edit"
              value="Redux"
            />
          </li>
        </ul>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="main"
    >
      <div>
        <input
          class="toggle-all"
          id="toggle-all"
          type="checkbox"
        />
        <label
          for="toggle-all"
        >
          Mark all as complete
        </label>
      </div>
      <ul
        class="todo-list"
      >
        <li
          class=" "
        >
          <div
            class="view"
          >
            <input
              class="toggle"
              type="checkbox"
            />
            <label>
              React
            </label>
            <button
              class="destroy"
            />
          </div>
          <input
            class="edit"
            value="React"
          />
        </li>
        <li
          class=" "
        >
          <div
            class="view"
          >
            <input
              class="toggle"
              type="checkbox"
            />
            <label>
              Redux
            </label>
            <button
              class="destroy"
            />
          </div>
          <input
            class="edit"
            value="Redux"
          />
        </li>
      </ul>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`TodoList should create TodoList with empty todos 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="main"
      >
        <div>
          <input
            class="toggle-all"
            id="toggle-all"
            type="checkbox"
          />
          <label
            for="toggle-all"
          >
            Mark all as complete
          </label>
        </div>
        <ul
          class="todo-list"
        />
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="main"
    >
      <div>
        <input
          class="toggle-all"
          id="toggle-all"
          type="checkbox"
        />
        <label
          for="toggle-all"
        >
          Mark all as complete
        </label>
      </div>
      <ul
        class="todo-list"
      />
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
